crates/sui-types/src/sui_system_state.rs

```diff
@@ -263,14 +323,13 @@ impl SuiSystemState {
             .active_validators
```

Comment:
Very cool. Now each time we are accessing the validator metadata, we can simply call verify on the metadata before accessing them, then we would no longer need the individual functions like create_narwhal_pubkey

Reply:
If you need more refactoring as part of this PR, I'd appreciate some additional guidance as this part of the codebase is new to me and I don't want to do something unexpected. Alternatively, perhaps this could become a separate task.